### ISSUE TITLE
Improve gmpy/sage selection decision

### ISSUE THREAD

**Issue by devurandom** [fredrik-johansson/mpmath#308]
I have been told that mpmath gets into trouble, when both gmpy and sage are available on a system. There would be some environment variables that force mpmath to select either one, but that is not very helpful if both are installed system wide and users are unaware of this problem. Thus I would like the selection of gmpy vs. sage as a backend within mpmath to be improved.

**Comment by kiwifb** [comment]
Hi, that refers to a private email conversation we had probably 18 to 24 months ago :) https://github.com/fredrik-johansson/mpmath/commit/9c5b5ad0f58d73d973a0ff6425ebcdc28a71a52e solved the problem that defining MPMATH_NOSAGE was not preventing sage to be used but we were talking about switching to a model where you would need a variable to enable a particular backend rather than disabling it.

Anyway my sage-on-gentoo user is complaining that I made sage and gmpy exclusive of each other. And I did that because I had too many reports of mysterious problems when both gmpy and sage were installed at the same time. Happy to revisit the situation if it shouldn't happen anymore.
